vaadin-testbench-unit-junit5/src/test/java/com/vaadin/testbench/unit/ClickableTest.java

```diff
@@ -0,0 +1,194 @@
+        Assertions.assertTrue(clickEvent.get().isShiftKey(),
+                "Shift key should be pressed");
+        Assertions.assertFalse(clickEvent.get().isAltKey(),
+                "Alt key should not be pressed");
+        Assertions.assertFalse(clickEvent.get().isMetaKey(),
+                "Meta key should not be pressed");
+    }
+
+    @Test
+    public void testMiddleClick() {
+        Div div = new Div("Middle click me");
+        container.add(div);
+
+        AtomicReference<Integer> buttonPressed = new AtomicReference<>();
+        div.addClickListener(e -> buttonPressed.set(e.getButton()));
+
+        ComponentTester<Div> divTester = test(div);
+        divTester.middleClick();
+
+        Assertions.assertEquals(1, buttonPressed.get(),
+                "Middle button (1) should be pressed");
+    }
+
+    @Test
+    public void testRightClick() {
+        Span span = new Span("Right click me");
+        container.add(span);
+
+        AtomicReference<Integer> buttonPressed = new AtomicReference<>();
+        span.addClickListener(e -> buttonPressed.set(e.getButton()));
+
+        ComponentTester<Span> spanTester = test(span);
+        spanTester.rightClick();
+
+        Assertions.assertEquals(2, buttonPressed.get(),
+                "Right button (2) should be pressed");
+    }
+
+    @Test
+    public void testClick_componentDisabled_throwsException() {
+        Button button = new Button("Disabled button");
+        button.setEnabled(false);
+        container.add(button);
+
+        ComponentTester<Button> buttonTester = test(button);
+
+        Assertions.assertThrows(IllegalStateException.class,
+                buttonTester::click,
+                "Should throw exception when clicking disabled component");
+    }
+
+    @Test
+    public void testClick_componentNotVisible_throwsException() {
+        Div div = new Div("Hidden div");
+        div.setVisible(false);
+        container.add(div);
+
+        ComponentTester<Div> divTester = test(div);
+
+        Assertions.assertThrows(IllegalStateException.class, divTester::click,
+                "Should throw exception when clicking invisible component");
+    }
+
+    @Test
+    public void testClick_componentNotAttached_throwsException() {
+        Span span = new Span("Detached span");
+        // Not adding to container, so it's not attached
+
+        ComponentTester<Span> spanTester = test(span);
+
+        Assertions.assertThrows(IllegalStateException.class, spanTester::click,
+                "Should throw exception when clicking detached component");
+    }
+
+    @Test
+    public void testClick_customComponent() {
+        // Custom component that doesn't implement ClickNotifier but handles
+        // click events directly
+        @Tag("custom-element")
+        class CustomComponent extends Component {
+            int clickCount = 0;
+
+            public CustomComponent() {
+                getEventBus().addListener(ClickEvent.class, e -> {
```

Comment:
It looks like this causes compilation errors without explicitly casting the lambda expression to `ComponentEventListener`.
not: it could maybe use `ComponentUtil.addListener`
```
                ComponentUtil.addListener(this, ClickEvent.class, (ComponentEventListener)  e -> {
                    clickCount++;
                });
```